backend/entities/Models/UserFilter.cs

```diff
@@ -0,0 +1,129 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Pims.Dal.Entities.Models
+{
+    /// <summary>
+    /// UserFilter class, provides a model for filtering user queries.
+    /// </summary>
+    public class UserFilter
```

Comment:
Provides a strongly typed list of what we want to allow users to filter/search/sort on.